### DISCUSSION TITLE
Change port of server without changing provision yaml

### DISCUSSION THREAD

**Discussion by LSnyd** [NVIDIA/NVFlare#442] [Q&A] [answered]
Hi all, 

I want to alter the port of admin and server in my script, so I can run multiple experiments on different ports on one node.

My project is set up similar to the cifar example. I found that I can alternate the port of the admin within the rn_secure.sh file in the example. However, I cannot find the location of the server port besides the one in the provision configuration yaml. Is there a way to change the value besides performing the provision every time from scratch?

**Comment by yhwen** [comment, marked as answer]
The FL server port and admin port are part of the NVFlare security set up. Those values are securely protected and can not be manually changed after the package provisioned. In order to set up the system using different ports, you need to update the project.yml and re-run the provisioning.

**Reply by LSnyd** [reply]
That's good to know, thank you!

**Reply by LSnyd** [reply]
@yhwen Is there a way to run multiple FL project instances on one node though? I tried to set up multiple secure workspaces with different ports and run them at the same time, but I keep getting communication errors.

**Reply by yhwen** [reply]
The current version of NVFlare 2.0 only supports single FL project. The multi-job Run support and HA solution will come out soon in the next 2.1 release.

**Reply by LSnyd** [reply]
Ok, looking forward to the next release then. What do you mean by HA solution though?

**Reply by yhwen** [reply]
High Availability for server failover.

**Reply by LSnyd** [reply]
Hi @yhwen, I created several secure workspace to try to add different ports. I deleted all of them in the end after you said that's not possible to do. However, I set up another secure workspace with the original settings, but now my project isn't running anymore and I am receiving the following error for server and admin: 
```
E0504 13:55:25.267134056   17173 ssl_transport_security.cc:1504] Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
E0504 13:55:26.273964158   17173 ssl_transport_security.cc:1504] Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
E0504 13:55:26.274471115   17173 ssl_transport_security.cc:1504] Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
E0504 13:55:26.777517058   17173 ssl_transport_security.cc:1504] Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
```

Do you have any idea where this may come from? Are there any old files from the previous workspaces that need to be deleted or so?

**Reply by nvkevlu** [reply]
I have encountered a similar error before and it happened when I had to install some updates for my machine which impacted SSL somehow. After rebooting, the error went away. Maybe there have been updates and your machine needs to reboot?

**Reply by yhwen** [reply]
It's probably you still have some previous running processes not been killed. Can you do a check and kill those background processes?

**Reply by LSnyd** [reply]
All other processes were killed. I've already checked that with `nvidia-smi` and `ps -ef|grep "projektname"` via the terminal.

**Reply by yhwen** [reply]
can you try `ps -aef | grep worker_process` ? (on the client)

**Reply by LSnyd** [reply]
I tried this as well, but there were no processes shown either. I just restarted our server and now it is running again. Seems like there were processes running, but they were not shown by any of these commands.
